dandiapi/api/services/asset/__init__.py

```diff
@@ -178,10 +181,18 @@ def add_asset_to_version(
             and asset_blob.embargoed
             and version.dandiset.embargo_status == Dandiset.EmbargoStatus.OPEN
         ):
-            AssetBlob.objects.filter(blob_id=asset_blob.blob_id).update(embargoed=False)
+            blob_id = asset_blob.blob_id
+            PublicAssetBlob.objects.filter(blob_id=blob_id).update(embargoed=False)
             transaction.on_commit(
-                lambda: remove_asset_blob_embargoed_tag_task.delay(blob_id=asset_blob.blob_id)
+                lambda: remove_asset_blob_embargoed_tag_task.delay(blob_id=blob_id)
             )
+        # TODO: add logic in case a public asset_blob identical to a private asset_blob is being uploaded
+        # in that case, save identical versions in public/private buckets
+        # maybe instead of elif statement add this to original if as an or condition?
+
+        # elif(asset_blob is not None
+        #      and settings.ALLOW_PRIVATE
+        #      and ): how to check for identical private asset?
```

Comment:
Suggested Edit: 
(copied some text from lines 175-176)

```suggestion
        # Question: Should creating an asset in an OPEN dandiset that points to a
        # private blob result in that blob becoming public?
        # If so, add logic to handle moving private asset_blob to public asset_blob

        # elif(asset_blob is not None
        #      and settings.ALLOW_PRIVATE
        #      and isinstance(value, PrivateAssetBlob):
```